internal/generate/olm-catalog/csv_updaters.go

```diff
@@ -226,39 +228,86 @@ var _ csvUpdater = crds{}
 		if err := yaml.Unmarshal(u, &crd); err != nil {
 			return err
 		}
 		for _, ver := range crd.Spec.Versions {
-			// Parse CRD descriptors from source code comments and annotations.
-			gvk := schema.GroupVersionKind{
-				Group:   crd.Spec.Group,
+			defKey := registry.DefinitionKey{
+				Name:    crd.GetName(),
 				Version: ver.Name,
 				Kind:    crd.Spec.Names.Kind,
 			}
-			newCRDDesc, err := descriptor.GetCRDDescriptionForGVK(scaffold.ApisDir, gvk)
-			if err != nil {
-				if goerrors.Is(err, descriptor.ErrAPIDirNotExist) {
-					log.Infof("Directory for API %s does not exist. Skipping CSV annotation parsing for API.", gvk)
-				} else if goerrors.Is(err, descriptor.ErrAPITypeNotFound) {
-					log.Infof("No kind type found for API %s. Skipping CSV annotation parsing for API.", gvk)
-				} else {
-					return fmt.Errorf("failed to set CRD descriptors for %s: %v", gvk, err)
-				}
-				continue
+			if owned, ownedExists := descMap[defKey]; ownedExists {
+				ownedDescs = append(ownedDescs, owned)
+			} else {
+				ownedDescs = append(ownedDescs, olmapiv1alpha1.CRDDescription{
+					Name:    defKey.Name,
+					Version: defKey.Version,
+					Kind:    defKey.Kind,
+				})
+			}
+		}
+	}
+	csv.Spec.CustomResourceDefinitions.Owned = ownedDescs
+	sort.Sort(descSorter(csv.Spec.CustomResourceDefinitions.Owned))
+	sort.Sort(descSorter(csv.Spec.CustomResourceDefinitions.Required))
+	return nil
+}
+
+func updateDescriptions(csv *olmapiv1alpha1.ClusterServiceVersion, searchDir string) error {
+	// TODO: Should this skip generating descriptions if API directory exists
+	// but project type can't be detected due to layout differences.
+	if opType := projutil.GetOperatorType(); opType != projutil.OperatorTypeGo {
+		log.Infof("Skipping generation of owned CRDDescriptions from CSV annotations in %s."+
+			" Could not detect project to be of Go type.", searchDir)
+		return nil
+	}
+
+	updatedDescriptions := []olmapiv1alpha1.CRDDescription{}
+	for _, currDescription := range csv.Spec.CustomResourceDefinitions.Owned {
+		group := currDescription.Name
+		if split := strings.Split(currDescription.Name, "."); len(split) > 1 {
+			group = strings.Join(split[1:], ".")
+		}
+		// Parse CRD descriptors from source code comments and annotations.
+		gvk := schema.GroupVersionKind{
+			Group:   group,
+			Version: currDescription.Version,
+			Kind:    currDescription.Kind,
+		}
+		newDescription, err := descriptor.GetCRDDescriptionForGVK(searchDir, gvk)
+		if err != nil {
+			if goerrors.Is(err, descriptor.ErrAPIDirNotExist) {
+				log.Infof("Directory for API %s does not exist. Skipping CSV annotation parsing for API.", gvk)
+			} else if goerrors.Is(err, descriptor.ErrAPITypeNotFound) {
+				log.Infof("No kind type found for API %s. Skipping CSV annotation parsing for API.", gvk)
+			} else {
+				// TODO: Should we ignore all CSV annotation parsing errors and simply log the error
+				// like we do for the above cases.
```

Comment:
```suggestion
// CSV annotation parsing errors will throw error. 
```
I think we could make a comment  here to make clear the current behaviour but the question could `should  we throw an error or just output warnings instead?` be raised in an issue. Then, IMO in this task we should first at all check what is the default behaviour in this scenario with kubebuilder/controller-tools and follow up it. WDYT?

Reply:
As mentioned above I'm going to file issues for these TODOs but I still want to keep them here so I can point to them later.
I think the line below `return fmt.Errorf("failed to set CRD descriptors for %s: %v", gvk, err)` makes it clear that we're returning an error so I don't think I need to add a comment here for that.